fastestimator/estimator.py

```diff
@@ -102,18 +79,15 @@ def fit(self):
         return self._start()
 
     def _prepare_pipeline(self):
-        # TODO - This needs to work with scheduling
         if self.steps_per_epoch is None:
-            pipeline = self.pipeline.get_current_value(epoch=0)
-            self.steps_per_epoch = pipeline.get_num_examples(mode="train", epoch=0) // pipeline.get_global_batch_size(
-                mode="train", epoch=0)
+            self.steps_per_epoch = self.pipeline.get_num_steps(mode="train", epoch=0)
         self.system.total_steps = self.epochs * self.steps_per_epoch
 
     def _prepare_network(self):
         self.network.exported_keys = self.network.op_outputs.intersection(self.trace_inputs)
 
     def _prepare_estimator(self):
-        self.do_eval = "eval" in self.pipeline.get_current_value(0).get_modes()  # TODO - Scheduling
+        self.do_eval = self.pipeline.eval_data is not None
```

Comment:
probably shouldn't reference internal variables like this directly. abstract class paradigm should define a relevant method instead